modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/UserResourceManagerTest.java

```diff
@@ -115,89 +364,865 @@ public void testGetSuccess(String id, String attributes,
+        SCIMResponse scimResponse = userResourceManager.create(scimObjectString,
+                userManager, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForTestCreateUserConflictException")
+    public Object[][] dataToTestCreateUserConflictException() {
+
+        return new Object[][]{
+                {NEW_USER_SCIM_OBJECT_STRING, "userName", null, ResponseCodeConstants.CODE_CONFLICT}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreateUserConflictException")
+    public void testCreateUserConflictException(String scimObjectString, String attributes,
+                                                String excludeAttributes, int expectedScimResponseStatus)
+            throws ConflictException, BadRequestException, CharonException {
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT);
+        abstractResourceManager.when(() -> AbstractResourceManager.encodeSCIMException(any(ConflictException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new ConflictException()));
+        Mockito.when(userManager.createUser(any(User.class), anyMap()))
+                .thenThrow(ConflictException.class);
+        SCIMResponse scimResponse = userResourceManager.create(scimObjectString, userManager,
+                attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+    }
+
+    @DataProvider(name = "dataForTestDeleteUserSuccess")
+    public Object[][] dataToTestDeleteUserSuccess()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        User user = getNewUser();
+        String id = user.getId();
+        return new Object[][]{
+                {id, ResponseCodeConstants.CODE_NO_CONTENT}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestDeleteUserSuccess")
+    public void testDeleteUserSuccess(String id, int expectedScimResponseStatus) {
+
+        abstractResourceManager.when(() -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_USER_ENDPOINT + "/" + USER_ID);
+        SCIMResponse scimResponse = userResourceManager.delete(id, userManager);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+        Assert.assertNull(scimResponse.getResponseMessage());
+        Assert.assertNull(scimResponse.getHeaderParamMap());
+    }
+
+    @DataProvider(name = "dataForTestDeleteUserFails")
+    public Object[][] dataToTestDeleteUserFails()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        User user = getNewUser();
+        String id = user.getId();
```

Comment:
```suggestion
        String id = getNewUser().getId();
```

Reply:
Fixed in commit 504369f8bdb4f024c63ae5d78f16b04decbedc11